python_hb_report/utillib.py

```diff
@@ -0,0 +1,924 @@
+			rc = 1
+		sys.stdout = oldout
+		nodes = myout.getvalue()
+		nodes = nodes.rstrip()
+		envir.USER_NODES = nodes.split('\n')
+		debug('Get CRM node list: '+' '.join(envir.USER_NODES))
+	except ValueError as msg:
+		rc = 1
+		msg.common_err(msg)
+	
+	return rc
+
+
+def get_nodes():
+	# 1. set bu user
+	if len(envir.USER_NODES):
+		print envir.USER_NODES
+	# 2. running cr,
+	elif iscrmrunning():
+		debug('querying CRM for nodes')
+		get_crm_nodes()
+		envir.NODE_SOURCE = 'crm'
+	# 3. hostcache
+	elif os.path.isfile(envir.HA_VARLIB+'/hostcache'):
+		utillib.debug('reading nodes from '+envir.HA_VARLIB+'/hostcache')
+		get_hostcache_node()
+		envir.NODE_SOURCE = 'hostcache'
+	# 4. ha.cf
+	elif envir.USER_CLUSTER_TYPE == 'heartbeat':
+		utillib.debug('reading node from ha.cf')
+		getcfvar('node')
+		envir.NODE_SOURCE = 'ha.cf'
+	# 5.of the cluster's stopped, try the CIB
+	elif os.path.isfile(envir.CIB_DIR+'/'+envir.CIB_F):
+		utillib.debug('reading node from the archived'+envir.CIB_DIR+'/'+envir.CIB_F)
+		CIB_file = os.path.join(envir.CIB_DIR,envir.CIB_F)
+		get_crm_node()
+		envir.NODE_SOURCE = 'crm'
+
+def do_which(command):
+	path = []
+	path = os.environ['PATH'].split(':')
+
+	for n in path:
+		dirlist = os.listdir(n)
+		if command in dirlist:
+			return True
+	
+	return False
+
+def do_greple(dirs,form):
+	'''
+	In dirs directoriy find file text  match form
+	Like grep -l -e, return the match name when get the first match 
+	'''
+	files = os.listdir(dirs)
+	log = ''
+	
+	for f in files:
+		#pass directires
+		if not os.path.isfile(f):			
+			continue
+
+		if f.find("ha-") != -1:
+			fd = open(f,'r')
+			txt = fd.readline()
+			#hit the targrt
+			while txt:
+				if txt.find(form) != -1:
+					return f
+				txt = fd.readline()
+			fd.close()
+
+	for f in files:
+
+		if not os.path.isfile(f):
+			continue
+
+		fd = open(f,'r')
+		txt = fd.readline()
+		#hit the targrt
+		while txt:
+			if txt.find(form) != -1:
+				return f
+			txt = fd.readline()
+		fd.close()
+		
+	return log
+
+def do_grep_file(files,form):
+	'''
+	In dirs directoriy find file text  match form
+	Like grep -l -e, return the match name when get the first match 
+	'''
+	log = ''
+		
+	if not os.path.isfile(files):
+		fatal(files+' is not exits')
+
+	fd = open(files,'r')
+	txt = fd.readline()
+	#hit the targrt
+	while txt:
+		if txt.find(form) != -1:
+			return txt
+		txt = fd.readline()
+	fd.close()
+
+	return ''
+		
+
+def findmsg():
+	'''
+	Found HA Log
+	'''
+	dirs="/var/log /var/logs /var/syslog /var/adm /var/log/ha /var/log/cluster /var/log/pacemaker /var/log/heartbeat /var/log/crm /var/log/corosync /var/log/openais"
+	syslogdirs = dirs.split()
+	favourites ='ha-*'
+	mark = envir.UNIQUE_MSG
+	log = []
+
+	for f in syslogdirs:
+		#grep pass directries
+		if not os.path.isdir(f):
+			continue
+		log = do_greple(f,mark)
+	
+	if not len(log):
+		debug('no HA log found in '+dirs)
+	else:
+		debug('found HA log at'+' '.join(log))
+	
+	return log
+
+
+def finf_getstampproc():
```

Comment:
Is that find_getstampproc?  If so, there is another dummy function named find_getstampproc